test/ripper/test_parser_events.rb

```diff
@@ -852,6 +852,38 @@ def test_lambda
     assert_equal true, thru_lambda
   end
 
+  def test_lambda_var
+    recv_params, recv_locals = nil, nil
+    parse('-> (foo) {}', :on_lambda_var) do |_, params, locals|
+      recv_params, recv_locals = params.list, locals
+    end
+
+    # parameters without block-local variables should return the list of
+    # parameters and false for locals
+    assert_equal ['foo'], recv_params
+    assert_equal false, recv_locals
+
```

Comment:
Needs no-paren case.
```suggestion

    recv_params, recv_locals = nil, nil
    parse('-> foo {}', :on_lambda_var) do |_, params, locals|
      recv_params, recv_locals = params.list, locals
    end

    # parameters without block-local variables should return the list of
    # parameters and false for locals
    assert_equal ['foo'], recv_params
    assert_equal false, recv_locals

```

I think we should have returned `nil`  for locals, but it may be too late...

Reply:
Yeah we have false for the other place too. In syntax tree I usually do `value || nil` to make sure it gets a `nil` anyway...